lucille-core/src/main/java/com/kmwllc/lucille/connector/storageclients/StorageClient.java

```diff
@@ -0,0 +1,92 @@
+package com.kmwllc.lucille.connector.storageclients;
+
+import static com.kmwllc.lucille.connector.FileConnector.AZURE_ACCOUNT_KEY;
+import static com.kmwllc.lucille.connector.FileConnector.GET_FILE_CONTENT;
+import static com.kmwllc.lucille.connector.FileConnector.GOOGLE_SERVICE_KEY;
+import static com.kmwllc.lucille.connector.FileConnector.S3_ACCESS_KEY_ID;
+import static com.kmwllc.lucille.connector.FileConnector.S3_SECRET_ACCESS_KEY;
+import static com.kmwllc.lucille.connector.FileConnector.S3_REGION;
+import static com.kmwllc.lucille.connector.FileConnector.AZURE_CONNECTION_STRING;
+import static com.kmwllc.lucille.connector.FileConnector.AZURE_ACCOUNT_NAME;
+
+import com.kmwllc.lucille.connector.FileConnector;
+import com.kmwllc.lucille.core.Publisher;
+import com.typesafe.config.Config;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Interface for cloud storage clients. Implementations of this interface should be able to connect to a cloud storage service
+ * and publish files to Lucille.
+ *
+ *  - getClient() - Get the appropriate client based on the URI scheme with authentication/settings from cloudOptions
+ *  - init() - Initialize the client
+ *  - shutdown() - Shutdown the client
+ *  - publishFiles() - Publish files to Lucille
+ *
+ */
+public interface StorageClient {
+  Logger log = LoggerFactory.getLogger(FileConnector.class);
+
+  static StorageClient getClient(URI pathToStorage, Publisher publisher, String docIdPrefix, List<Pattern> excludes, List<Pattern> includes,
```

Comment:
suggest moving this below the required methods

Reply:
resolved in [4250cb8](https://github.com/kmwtechnology/lucille/pull/217/commits/4250cb8b123ab6061697c25c193c021d3efc26d2)